docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,302 @@
+
+Example adding the migration to `init(self)`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  async init(self) {
+    // Adding a migration named 'add-featured-to-articles'
+    self.migration.add('add-featured-to-articles', async () => {
+      return self.apos.migration.eachDoc({
+        type: 'article'
+      }, async (doc) => {
+        if (doc.featured === undefined) {
+          await self.apos.doc.db.updateOne({
+            _id: doc._id
+          }, {
+            $set: { featured: false }
+          });
+        }
+      });
+    });
+  }
+};
+```
+<template v-slot:caption>
+  /modules/product/index.js
+</template>
+
+</AposCodeBlock>
+
+Example using `methods(self)`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  async init(self) {
+    self.apos.migration.add('add-product-description', self.addProductDescription);
+  }
+  methods(self) {
+    return {
+      async addProductDescription() {
+        await self.apos.migration.eachDoc({
+          type: 'product'
+        }, async (doc) => {
+          if (doc.description === undefined) {
+            await self.apos.doc.db.updateOne({
+              _id: doc._id
+            }, {
+              $set: { description: 'No description available' }
+            });
+          }
+        });
+      };
+    }
+  }
+};
+```
+<template v-slot:caption>
+  /modules/product/index.js
+</template>
+
+</AposCodeBlock>
+
+For both of these examples we are looping through all documents to find the `product` piece types. Then we are using the document `_id` and the Apostrophe database helper method `updateOne` to run the MongoDB operation `$set` that will either create or update the value of the `description` field for that piece. We will go through additional examples in detail below.
+
+### Running Migrations in Production
+
+In production environments, migrations are **not** automatically executed on every startup. This prevents accidental re-running of migrations in stable environments. Instead, migrations must be triggered manually by running a task.
+
+To apply pending migrations in a production environment, run:
+
+```bash
+node app @apostrophecms/migration:run
+```
+
+This ensures that any new migrations are executed in a controlled manner, which is particularly useful in environments where uptime and stability are critical.
+
+## Adding or Modifying a Property in Existing Documents
+
+When a new property needs to be added to all instances of a document type, you can use the [`eachDoc`](/reference/modules/migration.md#async-eachdoc-criteria-limit-iterator) helper provided by the migration module. This method efficiently queries documents in your collection and allows you to update them with only the necessary changes. The `eachDoc` helper takes three parameters.
```

Comment:
When a property of all instances of a document type needs to be changed, transformed or added in a way more complicated than setting `def` at the time it is first added to the code...

Reply:
Changed.